File: Implementation/상하좌우.java
package implementation;

import java.util.Scanner;

public class 상하좌우 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		System.out.printf("공간의 크기 N 입력>> ");
		int n = sc.nextInt();
		sc.nextLine();
		String[] plan = sc.nextLine().split(" ");

		int x = 1, y = 1;

		for (int i = 0; i < plan.length; i++) {
			switch (plan[i]) {
			case "L":
				if (y > 1)
					y--;
				break;
			case "R":
				if (y < n)
					y++;
				break;
			case "U":
				if (x > 1)
					x--;
				break;
			case "D":
				if (x < n)
					x++;
				break;
			default:
			}
		}
		System.out.printf("%d %d", x, y);
	}

}

File: python_for_coding_test/Implementation/상하좌우.py
def solution(n, plan):  # n: 공간의 크기, plan: 이동 계획서
    x, y = 1, 1
    d = {'L': [0, -1], 'R': [0, 1], 'U': [-1, 0], 'D': [1, 0]}  # 방향 맵핑
    for p in plan:
        x += d[p][0]
        y += d[p][1]
        if not 1 <= x <= n:
            x -= d[p][0]
        if not 1 <= y <= n:
            y -= d[p][1]

    return f'{x} {y}'


print(solution(5, ['R', 'R', 'R', 'U', 'D', 'D']))
